packages/flutter/lib/src/material/search.dart

```diff
@@ -0,0 +1,416 @@
+      primaryColor: Colors.white,
+      primaryIconTheme: theme.primaryIconTheme.copyWith(color: Colors.grey),
+      primaryColorBrightness: Brightness.light,
+      primaryTextTheme: theme.textTheme,
+    );
+  }
+
+  /// The current query string shown in the [Appbar].
+  ///
+  /// The user manipulates this string via the keyboard.
+  ///
+  /// If the user taps on a suggestion provided by [buildSuggestions] this
+  /// string should be updated to that suggestion via the setter.
+  String get query => _textEditingController.text;
+  set query(String value) {
+    assert(query != null);
+    _textEditingController.text = value;
+  }
+
+  /// Transition to show search results.
+  ///
+  /// If the user taps on a suggestion provided by [buildSuggestions] the
+  /// screen should typically transition to the page showing the search
+  /// results for the suggested query. This transition can be triggered
+  /// by calling this method.
+  ///
+  /// See also:
+  ///
+  ///  * [showSuggestions] to show the search suggestions again.
+  ///  * [currentBody] to learn whether the results or the suggestions are
+  ///    currently shown.
+  void showResults(BuildContext context) {
+    focusNode.unfocus();
+    _currentBody.value = SearchBody.results;
+  }
+
+  /// Transition to show the search suggestions and places the input focus
+  /// back into the search text field.
+  ///
+  /// If the results are currently shown this method can be used to go back
+  /// to showing the search suggestions.
+  ///
+  /// See also:
+  ///
+  ///  * [showResults] to show the search results.
+  ///  * [currentBody] to learn whether the results or the suggestions are
+  ///    currently shown.
+  void showSuggestions(BuildContext context) {
+    FocusScope.of(context).requestFocus(focusNode);
+    _currentBody.value = SearchBody.suggestions;
+  }
+
+  /// The body that is currently shown under the [AppBar] in the search page.
+  ///
+  /// Will be null if the search page is currently not shown.
+  ///
+  /// See also:
+  ///
+  ///  * [showResults] to show the search results.
+  ///  * [showSuggestions] to show the search suggestions.
+  SearchBody get currentBody => _currentBody.value;
+
+  /// Closes the search page and returns to the underlying route.
+  ///
+  /// The value provided for `result` is used as the return value of the call
+  /// to [showSearch] that launched the search initially.
+  void close(BuildContext context, T result) {
+    _currentBody.value = null;
+    focusNode.unfocus();
+    _result.complete(result);
+    Navigator.of(context).pop(result);
+  }
+
+  /// [Animation] triggered while the search pages fades in or out.
+  ///
+  /// This animation is commonly used to animate [AnimatedIcon]s of
+  /// [IconButton]s return by [buildLeading] or contained within the route
+  /// below the search page.
+  Animation<double> get transitionAnimation => _proxyAnimation;
+
+  /// [FocusNode] used by the text field showing the current search query.
```

Comment:
It's not really clear what this is for. In other words, if I'm building a SearchDelegate subclass what would I typically do with focusNode?

Reply:
Right. I'll make it private.